### ISSUE TITLE
Implement depth first mdp-tree expansion

### ISSUE THREAD

**Issue by KohlerHECTOR** [KohlerHECTOR/dpdt-py#1]
https://github.com/KohlerHECTOR/dpdt-py/blob/b56f6991bd32922181a0731df03bfc7595b6b4db/dpdt/dpdt_sklearn.py#L31 
We expand all the nodes at depth D. We should change that to do depth first expansion and directly backpropagate the reward on the expanded branch before expanding the new one.
